ciao-cli/tenant.go

```diff
@@ -17,16 +17,118 @@
 package main
 
 import (
+	"bytes"
+	"encoding/json"
+	"errors"
 	"flag"
 	"fmt"
 	"os"
 	"text/template"
 	"time"
 
+	"github.com/01org/ciao/ciao-controller/api"
 	"github.com/01org/ciao/ciao-controller/types"
 	"github.com/intel/tfortools"
 )
 
+func getCiaoTenantsResource() (string, error) {
+	url, err := getCiaoResource("tenants", api.TenantsV1)
+	return url, err
+}
+
+func getCiaoTenantRef(ID string) (string, error) {
+	var tenants types.TenantsListResponse
+
+	query := queryValue{
+		name:  "id",
+		value: ID,
+	}
+
+	url, err := getCiaoTenantsResource()
+	if err != nil {
+		return "", err
+	}
+
+	if !checkPrivilege() {
```

Comment:
Shouldn't this check go first?  Is there any point in calling getCiaoTenantsResource() if the user does not have the correct rights.  Also, we should return and error here.  This code will always return nil for the error.

Reply:
Maybe this check shouldn't even be here.  Wouldn't it prevent an unprivileged user from doing a 

ciao-cli tenant list --config

Reply:
@markdryan an unprivileged user is allowed to view their own config (but not modify it). This function is used for that purpose, as well as for privileged users. The check ensures that for a non privilege user the ref it gets is for a non-privileged endpoint, while for a privileged user the ref is for the privileged endpoint.

Reply:
The non-privileged user is prevented from doing ciao-cli tenant list --config for anyone except their own ID both programatically in ciao cli, and also in controller via a cert check. 
        if cmd.config {
                if checkPrivilege() == false {
                        return listTenantConfig(t, *tenantID)
                }

                return listTenantConfig(t, cmd.tenantID)
        }

Reply:
Okay, I think I see now.  If if the user isn't privileged we actually want to return the url we got from getCiaoTenantsResource.